weejson/src/main/scala/com/rallyhealth/weejson/v1/BufferedValue.scala

```diff
@@ -172,15 +186,57 @@ object BufferedValue extends Transformer[BufferedValue] {
   sealed trait AnyNum extends BufferedValue {
     def value: BigDecimal
   }
+
   case class Num(s: String, decIndex: Int, expIndex: Int) extends AnyNum {
-    override def value: BigDecimal = BigDecimal(s)
+    override lazy val value: BigDecimal = BigDecimal(s)
```

Comment:
This lazy val is going to cost an extra 8 bytes of allocation since it pushes us over the object alignment boundary. I'm not convinced that the extra memory footprint when buffering (and extra allocation pressure, even when memory is plentiful) is a good trade. It's fine to have a slower equals in `BufferedValue` since in practice, it should only be used for tests.

Reply:
I'll remove it, and we can study the performance tradeoffs later. My hope is that the stringy `Num` is something we can try to avoid in general (i.e., map to `Long` or `Double` early), but we gotta figure that out. My thoughts at the time I stuck in the `lazy` were (1) conversion from a `String` to `BigDecimal` is known to be expensive, (2) for the `Num` to be useful in a numeric context, it will have to get converted at some point, (3) if it's gonna happen once, we probably should cache the results somehow, and (4) putting in `lazy` will provoke a response from Doug and get us talking about it. Mission accomplished!